junit-platform-engine/src/main/java/org/junit/platform/engine/support/discovery/ResourceContainerSelectorResolver.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.engine.support.discovery;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toSet;
+import static org.junit.platform.commons.support.ReflectionSupport.findAllResourcesInClasspathRoot;
+import static org.junit.platform.commons.support.ReflectionSupport.findAllResourcesInPackage;
+import static org.junit.platform.commons.util.ReflectionUtils.findAllResourcesInModule;
+import static org.junit.platform.engine.support.discovery.ResourceUtils.packageName;
+import static org.junit.platform.engine.support.discovery.SelectorResolver.Resolution.selectors;
+import static org.junit.platform.engine.support.discovery.SelectorResolver.Resolution.unresolved;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.junit.platform.commons.support.Resource;
+import org.junit.platform.engine.discovery.ClasspathResourceSelector;
+import org.junit.platform.engine.discovery.ClasspathRootSelector;
+import org.junit.platform.engine.discovery.DiscoverySelectors;
+import org.junit.platform.engine.discovery.ModuleSelector;
+import org.junit.platform.engine.discovery.PackageSelector;
+
+/**
+ * @since 1.12
+ */
+class ResourceContainerSelectorResolver implements SelectorResolver {
+	private final Predicate<Resource> packageFilter;
+	private final Predicate<Resource> resourceFilter;
+
+	ResourceContainerSelectorResolver(Predicate<Resource> resourceFilter, Predicate<String> resourcePackageFilter) {
+		this.packageFilter = packageName(resourcePackageFilter);
+		this.resourceFilter = resourceFilter;
+	}
```

Comment:
Why are there two `Predicate<Resource>` but only one is passed to the methods in `ReflectionSupport` and one is used to filter in `resourceSelectors`?

Reply:
Ah good one. We only need one predicate to pass to the `ReflectionSupport` methods  with no further filtering.

Done in [1d13082](https://github.com/junit-team/junit5/pull/3718/commits/1d13082aa6eac12388feb7284d0ca2db7afb885c).